src/macfloat.c

```diff
@@ -70,6 +71,16 @@ static void PrintMacfloatToBuf(char *buf, size_t bufsize, Double val, int precis
     }
     else {
         snprintf(buf, bufsize, "%.*" PRINTFFORMAT, precision, val);
+        if (strlen(buf) < bufsize) {
```

Comment:
This check is incorrect. Correct one follows, together with a comment that explains it (a good idea in such delicate code anyway).
```suggestion
        // check if a period is in the output; this is not always the case,
        // e.g. if the value is an integer
        if (strchr(buf, '.'))
            return; // everything is fine
        // we need to insert a '.'; either at the end, or before an exponent
        // (e.g. "7e10" -> "7.e10"). For this we need 1 extra byte of storage,
        // plus of course 1 byte for the string terminator; check if the
        // buffer is big enough
        if (strlen(buf) + 2 <= bufsize) {
```

Reply:
Of course in reality, our buffers are 1024 bytes and won't overflow, but I always prefer to code defensively... and better no bounds check than an incorrect one... of course even better to have a correct one ;-)